### DISCUSSION TITLE
Unity Replacement?

### DISCUSSION THREAD

**Discussion by wowfood** [PrismLibrary/Prism#2734] [WPF] [answered]
Saw in another discussion post that unity is dead.  I know it'll still work but wondering if there are plans in the PRISM team to replace unity as the default with another container such as the Microsoft ServiceLocator.

If there were a switch to ServiceLocator it might make it simpler to integrate other microsoft packages that rely on it, such as the way add logging is handled, and allow some interoperability between PRISM desktop applications API code, and using that same code (without prism) in web applications.

**Comment by dansiegel** [comment, marked as answer]
First of all we've been telling people they should use DryIoc and not Unity Container for at least the last 3-4 years. As for other containers, I do have one that I've looked at implementing as a secondary choice for people, but have not had the time to implement. Microsoft.Extensions.DependencyInjection unfortunately remains incompatible for Prism apps for a variety of reasons. 

- No support for automatic concrete type resolution 
- No ability to append the services once the ServiceProvider is built
- While we may be able to use the changes in Prism.Maui for ViewModel resolution, Named registrations would not be possible which could break some existing apps.

**Comment by wowfood** [comment]
Ah that's understandable then.  

Also didn't realize you'd gone away from recommending unity.  I'd been following the guides on pluralsight which still focus on the Unity container.

**Reply by brianlagunas** [reply]
Not sure what you've been watching on Pluralsight, but the latest Prism for WPF course uses DryIoC. The older courses use Unity, but you can tell those are over 10 years old 😄

**Reply by wowfood** [reply]
TBF it's been a while since I needed to view the whole course, I normally just double check how certain bits I rarely use work.  So I probably missed the updates to the actual setup bits.
